autogen/oai/client.py

```diff
@@ -158,6 +159,8 @@ def create(self, params: Dict[str, Any]) -> ChatCompletion:
         Returns:
             The completion.
         """
+        iostream = IOStream.get_default()
```

Comment:
This is using the default stream. Should this stream be configurable directly from the constructor or the `create` method?

Reply:
The default stream is not global, it is an instance of a [`ContextVar`](https://docs.python.org/3/library/contextvars.html). If you need to pass it down to every function down the call stack in your thread, you can just do this:
```python
with IOStream.set_default(iostream):
  some_code_calling_create_down_the_call_stack()
```

It will not affect other default streams in other threads/async call stacks. In servers, that's IMO the easiest way to support multiple input/output streams.

Reply:
This is important to answer @jackgerrits 's question earlier in the office hour. I think we want to add this into the documentation.

Reply:
yeah this is pretty interesting, today I learned!

Reply:
I will write extensive documentation once I add structured output, there are many things to explain and provide examples for.